### DISCUSSION TITLE
MultiTenancy with EventBus using @ConsumeEvent

### DISCUSSION THREAD

**Discussion by Cyberdoc** [quarkusio/quarkus#37180] [Q&A]
Hi,

I've a project, which uses the vert.x EventBus for the communication between multiple services (this works as expected). Now we have the tenant information within the messages we send around. 
I wanted to use a TenantResolver based on the message and hoped to use then the PersistenceUnitExtension to set the connection to the correct database using a TenantConnectionResolver (as my tenant information are dynamic and not stored in the configuration files).

For HTTP requests, the context of a request is stored in the RoutingContext and RequestScoped. But at the moment I don't see something similar for Events and have question marks in my mind. 

Is there something? Or is my thinking, how to solve that, totally wrong and I would need to forward the TenantId from the Event-Consumer to the Service and then to the Repository and solve it there somwhow?

Kind regards,
Michael

**Comment by kpagratis** [comment]
Did you ever find a solution? I'm in the same boat

**Comment by dominique-jansen** [comment]
Maybe you are looking for the Annotation `@ActivateRequestContext`? Adding this to your consumer-method should trigger your TenantResolver before the database is accessed. Just make sure to write your tenant into the local context and read it back in your TenantResolver.
